FrostCord/FrostCord
# FROS Kicking and Banning

# Kicking and Banning
Introduces kicking/banning users and paves the way for...*leaving servers*

Do **NOT** review until #61 is implemented.

# API Changes
- Adds context menu options to kick/ban users based on the `MANAGE_USERS` permissions
- When users are not in a server, their server profile is stripped and only the base info provided by their `profiles` entry is used to fill in:
  - MessageHeader
  - MiniProfile
- Adds a tab to the settings panel to view bans and unban users

# Testing
- Verify kicking a user removes them from the server
- Verify the messages the user has sent in chat still have a header, even after a refresh
- Verify their miniprofile still opens but has no role section nor nickname
- Verify the user can rejoin the server by an invite link
- Verify the user's miniprofile from an old message now has the EVERYONE role again and allows you to kick/ban them again
- Verify banning a user removes them from a server
- Verify the user can no longer rejoin the server
- Unban the user from settings
- Verify the user can join the server again
